src/nodes/NodeConnectionManager.ts

```diff
@@ -1659,14 +1660,64 @@ class NodeConnectionManager {
       throw new nodesErrors.ErrorNodeConnectionManagerConnectionNotFound();
     }
     // Need to make an authenticate request here. Get the connection and RPC.
+    let reader: ReadableStreamDefaultReader | undefined;
+    let writer: WritableStreamDefaultWriter | undefined;
```

Comment:
You're throwing away all type checking with this. Be sure to apply the proper type with `ReadableStreamDefaultReader<T>`. The default generic assumes `any`.